python/private/pypi/parse_requirements.bzl

```diff
@@ -292,6 +292,21 @@ def _add_dists(*, requirement, index_urls, logger = None):
         index_urls: The result of simpleapi_download.
         logger: A logger for printing diagnostic info.
     """
+
+    # Handle direct URLs in requirements
+    if hasattr(requirement.srcs, "url"):
+        url = requirement.srcs.url
+        _, _, filename = url.rpartition("/")
+        direct_url_dist = struct(
+            url = url,
+            filename = filename,
+            sha256 = requirement.srcs.shas[0] if requirement.srcs.shas else "",
+            yanked = False,
+        )
+
+        # TODO should be able to handle sdist by checking the filename extension
+        return [direct_url_dist], None
```

Comment:
Anything that does not have `.whl` extension could be thought as an `sdist`, IMHO.